### PR TITLE
OCM-4965: Secure store tweaks

### PR DESCRIPTION
**Pull request by tylercreller**
* Add test for invalid and empty keyrings
* Additional Error Standardization
* Add support that allows the caller to define a keyring to target
* Add error handling for denied Keychain access due to permissions

### CODE REVIEW

**Review by tirthct — APPROVED** [review]

File: authentication/securestore/main.go [hunk 1]
```diff
@@ -103,32 +118,42 @@ func UpsertConfigToKeyring(creds []byte) error {
...
 
 	err = ring.Remove(ItemKey)
+	if err != nil {
+		if err == keyring.ErrKeyNotFound {
+			// Ignore not found errors, key is already removed
+			return nil
+		}
+
+		if strings.Contains(err.Error(), "Keychain Error. (-25244)") {
+			return fmt.Errorf("%s\nThis application may not have permission to delete from the Keychain. Please check the permissions in the Keychain and try again", err.Error())
```
Comment (macgregor) [inline authentication/securestore/main.go:139]: Friendly reminder that when wrapping errors your should use "%w" rather than %s or %v to properly preserve the original error in case callers want it: https://gosamples.dev/wrap-unwrap-errors/
